app/layout.tsx

```diff
@@ -0,0 +1,120 @@
+import { Layout, Navbar, Footer } from "nextra-theme-docs";
+import Link from "next/link";
+import { Head, Search, Banner } from "nextra/components";
+import { getPageMap } from "nextra/page-map";
+import "nextra-theme-docs/style.css";
+import Logo from "@/components/icons/logo.svg";
+import LogoIcon from "@/components/icons/logo-icon.svg";
+import BannerContents from "@/components/banner";
+import Providers from "@/components/providers";
+import { TocCTA } from "@/components/cta";
+import type { Metadata, ResolvingMetadata } from "next";
+import { SpeedInsights } from "@vercel/speed-insights/next";
+import "./globals.css";
+
+import { default as OurLayout } from "@/components/layout";
+
+export const generateMetadata = async (
+  _props: unknown,
+  parentPromise: ResolvingMetadata,
+): Promise<Metadata> => {
+  const { description } = await parentPromise;
+  return {
+    metadataBase: new URL("https://authzed.com"),
+    title: {
+      default: "Authzed Docs",
+      template: "%s - Authzed Docs",
+    },
+    description: "Welcome to the SpiceDB and AuthZed docs site.",
+    openGraph: {
+      title: {
+        default: "Authzed Docs",
+        template: "%s - Authzed Docs",
+      },
+      description: description ?? undefined,
+    },
+    alternates: {
+      // This is how you say "the current page is the canonical one",
+      // using the metadataBase field.
+      canonical: "./",
+    },
+  };
+};
+
+export default async function RootLayout({ children }) {
+  const pageMap = await getPageMap();
+  const enableSearch =
+    process.env.NEXT_PUBLIC_ENABLE_SEARCH_BLOG_INTEGRATION === "true";
+
+  const navbar = (
+    <Navbar
```

Comment:
There's a bunch of components that we were reimplementing that we can now pull directly from Nextra. `Navbar` and `Footer` are examples.